main.py

```diff
@@ -83,6 +84,14 @@ def delete_book():
     else:
         ui.message('Book not found in list \n')
 
+    try:
```

Comment:
This change is also unrelated. The unbound local error you are seeing is a bug in get_book_by_id